Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -55,12 +55,17 @@ struct MessageAlert: JSAlertInfo {
 struct ConfirmPanelAlert: JSAlertInfo {
     let message: String
     let frame: WKFrameInfo
+    let completionHandler: (Bool) -> Void
 
     func alertController() -> JSPromptAlertController {
         // Show JavaScript confirm dialogs.
         let alertController = JSPromptAlertController(title: titleForJavaScriptPanelInitiatedByFrame(frame), message: message, preferredStyle: .alert)
-        alertController.addAction(UIAlertAction(title: .OKString, style: .default))
-        alertController.addAction(UIAlertAction(title: .CancelString, style: .cancel))
+        alertController.addAction(UIAlertAction(title: .OKString, style: .default) { _ in
+           self.completionHandler(true)
+        })
+        alertController.addAction(UIAlertAction(title: .CancelString, style: .cancel) { _ in
+            completionHandler(false)
```

Comment:
I find that curious the previous `default` needs self but not this one?
```suggestion
            self.completionHandler(false)
```